### DISCUSSION TITLE
loki row context should be more specific for non-label query

### DISCUSSION THREAD

**Discussion by taylorchu** [grafana/grafana#59343] [Observability]
Currently if the query is like `{app="app1", role="role1"} | unpack | json | __error__="" | field1="test" |= "term1" |= "term2"`.

The row context query is `{app="app1", role="role1"}`. This is really useless because it is not specific enough. We use `pack` because `field1` could have high cardinality.

__Expected behavior__

1. The row context query is `{app="app1", role="role1"} | unpack | json | __error__="" | field1="test"`
2. Both `term1` and `term2` are highlighted in the surrounding context. Basically all `|=` and `|~` terms are highlighted. For example if the query is `{app="app1", role="role1"} |~ "strange term" | unpack | json | __error__="" | field1="test" |= "term1" |= "term2"`, then we will highlight `strange term`, `term1`, and `term2`.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
